packages/devtools_app/lib/src/screens/logging/logging_screen_v2/logging_table_v2.dart

```diff
@@ -0,0 +1,309 @@
+  State<_LoggingTableProgress> createState() => _LoggingTableProgressState();
+}
+
+class _LoggingTableProgressState extends State<_LoggingTableProgress> {
+  final _progressStopwatch = Stopwatch();
+  static const _millisecondsUntilCacheProgressShows = 500;
+  static const _millisecondsUntilCacheProgressHelperShows = 2000;
+
+  @override
+  Widget build(BuildContext context) {
+    return LayoutBuilder(
+      builder: (context, constraints) {
+        widget.model.tableWidth = constraints.maxWidth;
+        _progressStopwatch.reset();
+        return ValueListenableBuilder(
+          valueListenable: widget.model.cacheLoadProgress,
+          builder: (context, cacheLoadProgress, _) {
+            if (cacheLoadProgress != null) {
+              double progress = cacheLoadProgress;
+              if (!_progressStopwatch.isRunning) {
+                _progressStopwatch.start();
+              }
+
+              if (_progressStopwatch.elapsedMilliseconds <
+                  _millisecondsUntilCacheProgressShows) {
+                progress = 0.0;
+              }
+
+              return Center(
+                child: Column(
+                  mainAxisAlignment: MainAxisAlignment.center,
+                  children: [
+                    const Text(
+                      'Resizing... this will only take a moment.',
+                    ),
+                    const SizedBox(height: defaultSpacing),
+                    SizedBox(
+                      width: defaultLinearProgressIndicatorWidth,
+                      height: defaultLinearProgressIndicatorHeight,
+                      child: LinearProgressIndicator(
+                        value: progress,
+                      ),
+                    ),
+                    const SizedBox(height: defaultSpacing),
+                    if (_progressStopwatch.elapsedMilliseconds >
+                        _millisecondsUntilCacheProgressHelperShows)
+                      const Text(
+                        'To make this process faster, reduce the "Log Retention" setting.',
+                      ),
+                  ],
+                ),
+              );
+            } else {
+              _progressStopwatch
+                ..stop()
+                ..reset();
+              return _LoggingTableRows(model: widget.model);
+            }
+          },
+        );
+      },
+    );
+  }
+
+  @override
+  void dispose() {
+    super.dispose();
+    _progressStopwatch.stop();
+  }
+}
+
+class _LoggingTableRows extends StatefulWidget {
+  const _LoggingTableRows({
+    required this.model,
+  });
+
+  final LoggingTableModel model;
+
+  @override
+  State<_LoggingTableRows> createState() => _LoggingTableRowsState();
+}
+
+class _LoggingTableRowsState extends State<_LoggingTableRows> {
+  late final ScrollController _verticalController = ScrollController();
+
+  void onModelUpdate() {
+    setState(() {});
+  }
+
+  @override
+  void initState() {
+    widget.model.addListener(onModelUpdate);
```

Comment:
use the addAutoDisposeListener instead of manually adding and removing the listener. You will also be able to get rid of the onModelUpdate callback by doing this since this is the default when addAutoDisposeListener is not given a callback